### PR TITLE
Bookmarks page

### PR DESCRIPTION
**Pull request by foolhardy21**
## Bookmarks Page.

This PR has APIs and UI components related to bookmarks page. 

- Files to review:
   - [ ] src/reducers/bookmarks.reducer.js
   - [ ] src/pages/Bookmarks.jsx
   - [ ] src/contexts/bookmarks.context.js

- Use this link for preview - https://deploy-preview-6--remarkable-flan-74a807.netlify.app/login

v-1

Thanks

### CODE REVIEW

File: src/components/Explore/ExplorePost.jsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+            <p className='txt-secondary txt-md txt-500'>{'@ '}{username}</p>
+
+            <p className='txt-secondary txt-md txt-300 mg-left-xs mg-top-s mg-btm-s'>{content}</p>
+
+            <p className='txt-off-secondary txt-md txt-300'>{getDate(createdAt)}</p>
+
+            <p className='txt-off-secondary txt-md txt-300'>{getTime(createdAt)}</p>
+
+            <div className='flx flx-maj-stretch mg-top-s'>
```
Comment (nayyyhaa) [inline src/components/Explore/ExplorePost.jsx]: missing code formatter/prettier. You can remove extra lines from here.

File: src/utils/date.util.js [hunk 2]
```diff
@@ -19,4 +19,4 @@ export const getDate = date => `${new Date(date).getDate()}-${getMonthInWords(ne
  * @param {Date} date - date object
  * @returns {String} - time in HH:MM:SS format 
  */
-export const getTime = date => `${new Date(date).getHours()}:${new Date(date).getMinutes()}:${new Date(date).getSeconds()}`
\ No newline at end of file
+export const getTime = date => `${new Date(date).getHours()}:${new Date(date).getMinutes()}:${new Date(date).getSeconds()}`
```
Comment (nayyyhaa) [inline src/utils/date.util.js:22]: good that you've seperated timeFormate logic seperately.

**Review by nayyyhaa — APPROVED** [review]
Code is very well written & is easy to understand. You've broken all the logic separately for reusability. UI also looks clean & simple. You can go ahead & merge the PR. 💯

File: src/components/Explore/ExplorePost.jsx (hunk 1 above)
Comment (foolhardy21) [reply src/components/Explore/ExplorePost.jsx]: Sure, ill improve this.

File: src/components/Explore/ExplorePost.jsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+const ExplorePost = ({ post: { _id, username, content, likes: { likeCount }, createdAt } }) => {
+    const { isUserLoggedIn } = useAuth()
+    const { likePost, dislikePost, bookmarkPost, postsDispatch, removeBookmarkFromPost } = usePosts()
+    const { bookmarksDispatch } = useBookmarks()
+
+    const handlePostBookmark = async () => {
+        if (isUserLoggedIn) {
+            const response = await bookmarkPost(_id)
+            console.log(response)
```
Comment (ApurvaSawant11) [inline src/components/Explore/ExplorePost.jsx]: You can remove this if not needed

**Review by ApurvaSawant11 — APPROVED** [review]
The code looks clean and readable. The bookmarks functionality is working properly. You can merge the PR

File: src/components/Explore/ExplorePost.jsx (hunk 3 above)
Comment (foolhardy21) [reply src/components/Explore/ExplorePost.jsx]: yes, thanks for reminding.

**Pull request merged**